### PR TITLE
relax length constraints

### PR DESCRIPTION
**Pull request by 9il**

### CODE REVIEW

File: std/experimental/ndslice/slice.d [hunk 1]
```diff
@@ -62,10 +62,6 @@ auto sliced(
     if (!isStaticArray!Range && !isNarrowString!Range && N)
 in
 {
-    foreach (len; lengths)
-        assert(len > 0,
```
Comment (JackStouffer) [inline std/experimental/ndslice/slice.d]: Why not just make it 

```
assert(len >= 0,
```

to  allow for zero length slices while still keeping the check?

Comment (9il) [inline std/experimental/ndslice/slice.d]: `size_t` is always not negative

Comment (JackStouffer) [inline std/experimental/ndslice/slice.d]: right

**Comment by DmitryOlshansky** [comment]
LGTM

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Comment by 9il** [comment]
Thanks!

**Pull request merged**